pkg/server/container_create_test.go

```diff
@@ -610,6 +576,115 @@ func TestPrivilegedBindMount(t *testing.T) {
 	}
 }
 
+func TestMountPropagation(t *testing.T) {
+	sharedGetMountsFn := func() ([]*mount.Info, error) {
+		return []*mount.Info{
+			{
+				Mountpoint: "host-path",
+				Optional:   "shared:",
+			},
+		}, nil
+	}
+
+	slaveGetMountsFn := func() ([]*mount.Info, error) {
+		return []*mount.Info{
+			{
+				Mountpoint: "host-path",
+				Optional:   "master:",
+			},
+		}, nil
+	}
+
+	othersGetMountsFn := func() ([]*mount.Info, error) {
+		return []*mount.Info{
+			{
+				Mountpoint: "host-path",
+				Optional:   "others",
+			},
+		}, nil
+	}
+
+	for desc, test := range map[string]struct {
+		criMount        *runtime.Mount
```

Comment:
Add `expectErr bool`.

We could:
1) assert error, if `expectErr==true`.
2) assert no error and check options, if `expectErr==false`.

Reply:
OK, I will change according to your instructions, thanks!

Reply:
Done!